src/python/zquantum/core/cost_function.py

```diff
@@ -5,11 +5,47 @@
 from .interfaces.estimator import Estimator
 from .circuit import combine_ansatz_params
 from .estimator import BasicEstimator
+from .measurement import ExpectationValues
 from typing import Optional
 import numpy as np
 from openfermion import SymbolicOperator
 
 
+def sum_expectation_values(expectation_values: ExpectationValues) -> ValueEstimate:
+    """Compute the sum of expectation values.
+
+    If correlations are available, the precision of the sum is computed as
+
+    \epsilon = \sqrt{\sum_k \sigma^2_k}
+
+    where the sum runs over frames and \sigma^2_k is the estimated variance of
+    the estimated contribution of frame k to the total. This is calculated as
+
+    \sigma^2_k = \sum_{i,j} Cov(o_{k,i}, o_{k, j})
+
+    where Cov(o_{k,i}, o_{k, j}) is the estimated covariance in the estimated
+    expectation values of operators i and j of frame k.
+
+    Args:
+        expectation_values: The expectation values to sum.
+    
+    Returns:
+        The value of the sum, including a precision if the expectation values
+            included covariances.
+    """
+
+    value = np.sum(expectation_values.values)
+
+    precision = None
+
+    if expectation_values.covariances:
+        variance = 0
+        for frame_covariance in expectation_values.covariances:
+            variance += np.sum(frame_covariance, (0, 1))
+        precision = np.sqrt(variance)
```

Comment:
`precision` is a sqrt of the estimated variance of the ising operator but we actually need an estimate of the standard error (https://en.wikipedia.org/wiki/Standard_error). @jgonthier

Reply:
It actually is the standard error: `expectation_values.covariances` is the covariances of the estimators for the operator expectation values, not the covariances of the operators. I've renamed `covariances` to `estimator_covariances` and `variances` to `estimator_variance` to make this less confusing!

Reply:
Okay, gotcha. Sorry about that,  I got a bit confused!